cypress/integration/search_page_spec.js

```diff
@@ -0,0 +1,207 @@
+    response: 'fixture:search_page_spec/cogo_datasets'
+  },
+  cotaDatasets: {
+    method: 'GET',
+    url: '/api/v1/dataset/search?offset=0&limit=10&sort=name_asc&query=COTA&apiAccessible=true',
+    response: 'fixture:search_page_spec/cota_datasets'
+  },
+  info: {
+    method: 'GET',
+    url: '/sockjs-node/*',
+    response: 'fixture:search_page_spec/info.json'
+  },
+  ogripDataset: {
+    method: 'GET',
+    url: '/api/v1/organization/ogrip/dataset/622746a5_4e2a_4a4c_ac18_74cb1fb05ab3',
+    response: 'fixture:search_page_spec/ogrip_dataset'
+  }
+}
+
+function isDefaultPage () {
+  cy.get(organizations).children('.checkbox').should('have.length', 8)
+  cy.get(keywords).children('.checkbox').should('have.length', 10)
+  cy.get(datasetsFoundCount).contains('19 datasets found')
+  cy.get(datasets).find('data-card').should('have.length', 10)
+  cy.get(paginator).find('button.page-number').should('have.length', 2)
+  cy.get(sortSelectBox).should('have.value', 'name_asc')
+  cy.get(apiAccessibleCheckbox).should('have.class', 'selected')
+  cy.get(firstDataset).contains('100 Year Flood Plain -- GeoJSON')
+}
+
+function isCoGoPage () {
+  cy.url().should('contain', 'organization%5D%5B%5D=COGO&')
+  cy.get(organizations).children('.checkbox').should('have.length', 1)
+  cy.get(keywords).children('.checkbox').should('have.length', 6)
+}
+
+function isBicyclePage () {
+  cy.url().should('contain', 'keywords%5D%5B%5D=bicycle')
+  cy.get(organizations).children('.checkbox').should('have.length', 1)
+  cy.get(keywords).children('.checkbox').should('have.length', 6)
+}
+
+function isFacetList () {
+  cy.get(dialogContent).find('.section-header').contains('keywords')
+  cy.get(dialogContent).find('.section').children('.checkbox').should('have.length', 46)
+}
+
+describe('Test search interactions on the page', function () {
+  beforeEach(function () {
+    cy.server()
+    cy.route(routes.allDatasetsNameAsc)
+    cy.route(routes.info)
+    cy.visit('/')
+  })
+
+  it('successfully loads', function () {
+    isDefaultPage()
+  })
+
+  it('search works', function () {
+    cy.route(routes.cotaDatasets)
+    cy.get(search).type('COTA{enter}')
+    cy.get(datasetsFoundCount).contains('1 datasets found for "COTA"')
+    cy.contains('COTA Real Time Bus Locations')
+  })
+
+  it('sort works', function () {
+    cy.route(routes.allDatasetsNameDesc).as('getDatasetsInDescendingOrderByName')
+    cy.route(routes.allDatasetsLastModified).as('getDatasetsByLastModifiedDate')
+    cy.get(sortSelectBox).select('name_desc')
+    cy.wait(['@getDatasetsInDescendingOrderByName'])
+    cy.get(sortSelectBox).should('have.value', 'name_desc')
+    cy.get(firstDataset).contains('Sample XML Dataset - DEV')
+    cy.get(sortSelectBox).select('last_mod')
+    cy.wait(['@getDatasetsByLastModifiedDate'])
+    cy.get(sortSelectBox).should('have.value', 'last_mod')
+    cy.get(firstDataset).contains('COTA Real Time Bus Locations')
+  })
```

Comment:
(Tony) Would it be possible to break tests like this into multiple describe/beforeEach/it blocks?  Something like would be a lot more descriptive of what you're testing:

```
  describe('sort', function () {
    beforeEach(function () {
      cy.route(routes.allDatasetsNameDesc).as('getDatasetsInDescendingOrderByName')
      cy.route(routes.allDatasetsLastModified).as('getDatasetsByLastModifiedDate')
    })

    it('can sort by name descending' function () {
      cy.get(sortSelectBox).select('name_desc')
      cy.wait(['@getDatasetsInDescendingOrderByName'])
      cy.get(sortSelectBox).should('have.value', 'name_desc')
      cy.get(firstDataset).contains('Sample XML Dataset - DEV')
    })

    it('can sort by last modified date' function () {
      cy.get(sortSelectBox).select('last_mod')
      cy.wait(['@getDatasetsByLastModifiedDate'])
      cy.get(sortSelectBox).should('have.value', 'last_mod')
      cy.get(firstDataset).contains('COTA Real Time Bus Locations')
    })
  })
```

You could take this even further by  putting each `should` or `contains` in its own `it` block, but hopefully you get the idea.

Reply:
If this was a unit test I would agree with you, but I think that this might be a little too granular according for [Cypress](https://docs.cypress.io/guides/references/best-practices.html#Creating-%E2%80%9Ctiny%E2%80%9D-tests-with-a-single-assertion).